File: BackEnd/eMenka/eMenka.Domain/Classes/Supplier.cs
using eMenka.Domain.Enums;

namespace eMenka.Domain.Classes
{
    public class Supplier
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public SupplierType[] Types { get; set; }
        public bool Active { get; set; }
        public bool Internal { get; set; }
    }
}

File: eFenkaFrontend/src/app/models/supplier/supplier.ts
import { SupplierType } from '../../enums/supplier-type/supplier-type.enum';

export class Supplier {
  id: number;
  name: string;
  active: boolean;
  internal: boolean;
  types: number[];

  constructor(id: number, name: string, active: boolean, internal: boolean, types: SupplierType[]) {
    this.id = id;
    this.name = name;
    this.active = active;
    this.internal = internal;
    this.types = types;
  }
}
